File: rabbit_mq/src/main/java/com/rabbit/rabbit_mq/domain/chat/entity/ChatRoom.java
package com.rabbit.rabbit_mq.domain.chat.entity;

import com.rabbit.rabbit_mq.domain.member.entity.Member;
import com.rabbit.rabbit_mq.global.jpa.BaseTime;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import lombok.*;

import static lombok.AccessLevel.PROTECTED;

@Entity
@Builder
@NoArgsConstructor(access = PROTECTED)
@AllArgsConstructor(access = PROTECTED)
@Getter
@Setter
public class ChatRoom extends BaseTime {
    @ManyToOne
    private Member owner;
    private String name;
    private boolean published;
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/domain/chat/entity/ChatRoom.kt
package com.krstudy.kapi.domain.chat.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.krstudy.kapi.domain.member.entity.Member
import com.krstudy.kapi.global.jpa.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToOne

@Entity
@JsonIgnoreProperties(ignoreUnknown = true)
data class ChatRoom(

    @Column(name = "room_name")
    val roomName: String? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    var author: Member? = null,

    ) : BaseEntity() {}